File: src/main/kotlin/org/kite9/diagram/visualization/orthogonalization/flow/balanced/TerminatorPair.java
package org.kite9.diagram.visualization.orthogonalization.flow.balanced;

import org.kite9.diagram.common.objects.Pair;
import org.kite9.diagram.model.Terminator;

public class TerminatorPair extends Pair<Terminator>{

	public TerminatorPair(Terminator a, Terminator b) {
		super(a, b);
	}

	@Override
	protected boolean elementEquals(Object a2, Object a3) {
		return ((Terminator)a2).styleMatches((Terminator)a3);
	}

	
}

File: visualization/kite9-visualization-common/src/jvmMain/kotlin/org/kite9/diagram/visualization/orthogonalization/flow/balanced/TerminatorPair.kt
package org.kite9.diagram.visualization.orthogonalization.flow.balanced

import org.kite9.diagram.common.objects.Pair
import org.kite9.diagram.model.Terminator

class TerminatorPair(a: Terminator?, b: Terminator?) : Pair<Terminator?>(a, b) {

    override fun elementEquals(a2: Any, a3: Any): Boolean {
        return (a2 as Terminator?)?.styleMatches((a3 as Terminator?)) ?: true
    }
}